design/specs/add-formula-column.md

```diff
@@ -0,0 +1,180 @@
+---
+title: Listing Views Spec
+description: 
+published: true
+date: 2022-06-20T08:24:04Z
+tags: 
+editor: markdown
+dateCreated: 2022-06-20T08:24:04Z
+---
+
+## Context
+
+In addition to data columns, we have formula columns as well. These columns are basically derived from a formula. A formula may or may not use another data column to generate the formula column. The formula can also use parameters of different data types other than columns. 
+
+The formula columns can only be created with Data explorer. For more details, you can check out this [link](https://wiki.mathesar.org/en/product/specs/2022-01-views/04-formulas)
+
+## Scenarios
+
+### Scenario 1 - Add a formula column without clicking outside of the inspector.
+
+![](https://share.balsamiq.com/c/dS6sAay4Re9f2R39zKDDj8.png)
+
+On the left hand side of the explorer view, there is a  `Add Column`  button. When user clicks on it, there are two options visible to them. One is `Add direct column` and the other is `Add formula column`. To add a formula column user will need to click on `Add formula column`. 
+
+Once the user clicks on `Add formula column` option, the inspector shows formula settings that include `Name` input of the formula and a list of formulas to select from. The list of formulas is described in detail in the [formula selection menu](#components) section. 
+
+If the name is left empty by user for the next steps, there is a default name given to the column that is FormulaX. X is the number given to the column. For instance if its the first default named column it will be Formula1.
+
```

Comment:
I think the name will be set by default at the moment of column creation. The user can choose to change it.

Reply:
Yes.